esp-hal/src/timer/mod.rs

```diff
@@ -105,22 +104,40 @@ pub trait Timer: crate::private::Sealed {
     /// Has the timer triggered?
     fn is_interrupt_set(&self) -> bool;
 
+    /// Asynchronously wait for the timer interrupt to fire.
+    ///
+    /// Requires the correct `InterruptHandler` to be installed to function
```

Comment:
I deliberately decoupled the async fn from installing the correct interrupt handler, as this is a low level trait and if a user is reaching for it they probably want to register there own handler which could also call the async one once they're done doing what they need to do with it.